build.gradle

```diff
@@ -1,114 +1,147 @@
 /// Gradle Build script for 18xx Rails program (rails.sf.net)
 
-import org.apache.tools.ant.filters.*
 import java.text.SimpleDateFormat
 
-
-
-// Addition of JavaProfile plugin to read Java property files
-// reference: https://github.com/unsaved/gradle-javaPropFile-plugin
-
-buildscript {
-    repositories { jcenter() }  // Before plugin v. 1.0.0 use mavenCentral()
-    dependencies {
-        classpath 'com.admc:gradle-javaPropFile-plugin:1.0.0'
-        // For strictly repeatable builds, replace 'latest.integration'
-        // with the latest literal GA version label.
-    }
-}
-
-
-// Addition of launch4j gradle plugin
-// refer to https://github.com/TheBoegl/gradle-launch4j
-
-buildscript {
-  repositories {
-    maven {
-      url "https://plugins.gradle.org/m2/"
-    }
-  }
-  dependencies {
-    classpath "gradle.plugin.edu.sc.seis.gradle:launch4j:2.3.0"
-  }
-}
-
-// Addition of shadow plugin to create one jars
-// reference: https://github.com/johnrengelman/shadow
-plugins {
-  id 'java' // or 'groovy' Must be explicitly applied
-  id 'com.github.johnrengelman.shadow' version '5.0.0'
-}
-
-// Addition of macAppBundle
-// https://github.com/crotwell/gradle-macappbundle
 plugins {
+  id 'java'
+  id 'maven-publish'
+  // https://github.com/johnrengelman/shadow
+  id 'com.github.johnrengelman.shadow' version '5.2.0'
+  // plugin to check for dependency updates via task dependencyUpdates
+  // https://github.com/ben-manes/gradle-versions-plugin
+  id 'com.github.ben-manes.versions' version '0.28.0'
+  // https://github.com/n0mer/gradle-git-properties
+  id 'com.gorylenko.gradle-git-properties' version '2.2.2'
+  // https://github.com/TheBoegl/gradle-launch4j
+  id 'edu.sc.seis.launch4j' version '2.4.6'
+  // https://github.com/crotwell/gradle-macappbundle
   id "edu.sc.seis.macAppBundle" version "2.3.0"
+  // https://github.com/nebula-plugins/gradle-ospackage-plugin
+  //id "nebula.rpm" version "8.2.0"
+  //id "nebula.deb" version "8.2.0"
+  id "nebula.ospackage" version "8.2.0"
+  // https://openjfx.io/openjfx-docs/#gradle
+  id 'org.openjfx.javafxplugin' version '0.0.8'
+  id "jacoco"
+  id "org.sonarqube" version "2.8"
 }
 
-
 repositories {
-  mavenCentral()
+    mavenLocal()
+    mavenCentral()
+//    maven {
+//        url = 'https://jitpack.io'
+//    }
 }
 
-apply plugin: 'java'
-sourceCompatibility = 1.6
-targetCompatibility = 1.6
-
-apply plugin: "edu.sc.seis.launch4j"
-
-apply plugin: 'com.admc.javaPropFile'
+publishing {
+    publications {
+        maven(MavenPublication) {
+            from(components.java)
+        }
+    }
+}
```

Comment:
What does this do?

Reply:
Used to publish the artifacts to maven central. Don't know how we publish today so this likely would be changed

Reply:
As far as I know we don't publish the artifacts to a repository. The most we do currently is to link them on the rails website: https://rails.sourceforge.io/

Long-term I would like to use the `Releases` page on GitHub to host our releases. Maybe even add a GutHub Page as a website.

Reply:
The sourceforge release is based upon legacy expectations management. I released the last two builds also on Github. But since most ppl in the past looked for Sourceforge we need to publish the packages on both sites.

On a different Note. Please guys also use the Slack Channel as not all guys interested in Rails development are current following the issues and pull requests here :)

Reply:
@neutronc I don't seem to have a link to the Slack channel. Can you please post one? In addition I think it would be good to add the link to the README as well. Maybe behind a nice Button image